plugins/pulp_rpm/plugins/importers/yum/upload.py

```diff
@@ -186,19 +186,18 @@ def _handle_erratum(repo, type_id, unit_key, metadata, file_path, conduit, confi
     unit_data.update(metadata or {})
     unit_data.update(unit_key or {})
 
-    existing_unit = model_class.objects.filter(**unit_key).first()
-    new_unit = model_class(**unit_data)
+    unit = model_class(**unit_data)
+    unit._repo_id = repo.repo_id
```

Comment:
I think in general errata as a whole is imported to some repo and not pkglist only so it makes some sense to add reponame to errata itself. I do not feel strongly about both solutions though.

To elaborate a bit on your suggestion. 
Pkglist can contain multiple collections and in this case pkglist looks like this:
```
pkglist: [
  {
    name: reponame,
    packages: [{...}, {...}]
  },
  {
    name: reponame,
    packages: [{...}, {...}]
  },
  ...
]
```
Reponame is the same in every collection, so it is redundant. If we do exactly the way we did before, we will add reponame to every collection during import (before saving erratum unit) and we will have a structure as shown above. Then when we create and save pkglist into db we will need either to save this redundant and no longer needed reponame  into db or remove it from every collection. That's whyI wanted to have reponame only in one place and I chose Errata for that. 
Another option is that pkglist can be temporarily (in memory) changed to something like this is:
```
pkglist: {
  'repo_id': reponame,
  'collections':  [
    {
      packages: [{...}, {...}]
    },
    {
      packages: [{...}, {...}]
    },
    ...
]
```
Maybe Errata model is still the best of two evils, what do you think?